src/class-database-size-plugin.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+* Get the sizes of all tables in the Database
+* and display them
+* @author Matthias <matthias@savvii.com>
+*/
+
+namespace Savvii;
+
+class DatabaseSizePlugin {
+
+    function __construct() {
+        add_action( 'admin_menu', [ $this, 'admin_menu_init' ]);
+        add_action( 'admin_bar_menu', [ $this, 'admin_bar_menu' ], 90);
+    }
+
+    function admin_menu_init() {
+        add_submenu_page(
+            'warpdrive_dashboard',
+            'Database information',
+            'Database information',
+            'manage_options',
+            'warpdrive_databaseinfo',
+            [ $this, 'databaseinfo_page']
+        );
+    }
+
+    function admin_bar_menu() {
+        global $wp_admin_bar;
+
+        if ( is_super_admin() ) {
+            $wp_admin_bar->add_menu([
+                'parent' => 'warpdrive_top_menu',
+                'id' => 'warpdrive_databaseinfo',
+                'title' => 'Database information',
+                'href' => wp_nonce_url( admin_url( 'admin.php?page=warpdrive_databaseinfo' ), 'warpdrive_databaseinfo' ),
+            ]);
+        }
+    }
+
+    function databaseinfo_page() {
+        check_admin_referer( 'warpdrive_databaseinfo' );
+
+        $tables = Database::get_wp_table_sizes();
+
+        // get and check the database info
+        $databaseinfo = Database::get_wp_database_size();
+        $errormsg = '';
+        $databasename = '';
+        $databasesize = 0;
+        if (
+                !array_key_exists('database', $databaseinfo) ||
+                !array_key_exists('size', $databaseinfo) ||
+                empty($tables)
+        ) {
+            $errormsg = "Could not retrieve database information, please contact support";
```

Comment:
Nice to have: actual email address here.

Reply:
I'll add it